pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -271,26 +279,162 @@ func (rs *ReceiverService) DeleteReceiver(ctx context.Context, uid string, orgID
 		if err != nil {
 			return err
 		}
-		return rs.deleteProvenances(ctx, orgID, postable.GrafanaManagedReceivers)
+		return rs.deleteProvenances(ctx, orgID, existing.Integrations)
+	})
+}
+
+func (rs *ReceiverService) CreateReceiver(ctx context.Context, r *models.Receiver, orgID int64, user identity.Requester) (*models.Receiver, error) {
```

Comment:
I think we should perform struct validation before any other activities, i.e. validate against alerting integrations.